Goal: report failure to execute the shell to the user and to syslog.
      Also, return a 126 or 127 exit status when !USE_PAM (was mising
      from another patch.

Index: shadow-4.0.12/src/su.c
===================================================================
--- shadow-4.0.12.orig/src/su.c	2005-08-17 11:50:43.000000000 +0200
+++ shadow-4.0.12/src/su.c	2005-08-17 11:50:44.000000000 +0200
@@ -233,6 +233,13 @@
 
 #ifndef USE_PAM
 	(void) execv (shellstr, args);
+	fprintf (stderr, _("No shell\n"));
+	SYSLOG((LOG_WARN, "Cannot execute %s\n", shellstr));
+	{
+		int exit_status = (errno == ENOENT ? 127 : 126);
+
+		exit (exit_status);
+	}
 	/* NOT REACHED */
 #else
 	child = fork ();
@@ -240,6 +247,8 @@
 		pam_end (pamh, PAM_SUCCESS);
 
 		(void) execv (shellstr, (char **) args);
+		fprintf (stderr, _("No shell\n"));
+		SYSLOG((LOG_WARN, "Cannot execute %s\n", shellstr));
 		{
 			int exit_status = (errno == ENOENT ? 127 : 126);
 
